### ISSUE TITLE
Gateway quiz submission screen incorrectly displays quiz total for quizzes with 0-point items

### ISSUE THREAD

**Issue by awmorp** [openwebwork/webwork2#1201]
When a gateway quiz has a problem with a weight of 0, the score is not correctly reported on the submission screen:
![image](https://user-images.githubusercontent.com/8839416/108162944-45565c80-7142-11eb-9d55-65d2ab8631fe.png)
In this example, the submission screen shows "Your score on this test is 2/2" when it should be 1/1.

Looks like the bug is in these lines, which assume a default weight of 1:
https://github.com/openwebwork/webwork2/blob/03eca4cc1a6034d24cf28bd690f0f56ce14f284b/lib/WeBWorK/ContentGenerator/GatewayQuiz.pm#L1795
https://github.com/openwebwork/webwork2/blob/03eca4cc1a6034d24cf28bd690f0f56ce14f284b/lib/WeBWorK/ContentGenerator/GatewayQuiz.pm#L1811

Having problems with weight 0 is useful for displaying instructions at the start of the quiz, for instance.

**Comment by drgrice1** [comment]
I am incorrect with the weighting comment.  I forgot how it was displayed there.   If there are 2 problems and the first has a weight of 9 and the second a weight of 1, and a student only gets the second one correct, then the Grades page will show both 10% and 1/10 and the quiz it will show 1/10, and so there is no discrepancy.

So the question just comes down to how to handle the score of 0 consistently.

**Comment by drgrice1** [comment]
There is another place where the quiz score is displayed.  That is on the "Homework Sets" page.  There the score is displayed in the same was as in the gateway quiz itself.  So in your example it shows 2/2, and not 1/1.  So it seems those two are consistent, but the Grades page is not consistent with those.

**Comment by awmorp** [comment]
Ok, consider the case where the first question is purely information about the quiz and doesn't have any answer blanks and has weight 0. If the student answers the second question correctly, the submission screen shows 1/2, ie 50%, when actually they scored 100%. So 1/2 is clearly incorrect, and begs questions from students "why did I only get 50% when it says I got all answers correct?":
![image](https://user-images.githubusercontent.com/8839416/108274723-63b36b00-71c9-11eb-8ecc-d1e8666eba29.png)

**Comment by Alex-Jordan** [comment]
> consider the case where the first question is purely information about the quiz

I know this is a common practice, but this is a hack to work around a deficiency of the quiz module. There should be a place where a set header file can be displayed. And it should be visible even if you are on a page that does not show question #1.

For the long term solution to the issue here, I propose not considering this use case (and assuming that we can build a place to display set header information).

Do any use cases remain where a question would have no answer input fields?

**Comment by awmorp** [comment]
> For the long term solution to the issue here, I propose not considering this use case

Ok

> Do any use cases remain where a question would have no answer input fields?

Only thing I could think of is maybe as section headers, eg "Section 2: The following 5 questions relate to integral calculus". Also a bit hackish.

**Comment by drgrice1** [comment]
I agree that this use case is not worth attention.  I don't see the section header idea as being particularly valid either.

I would like to see a confirmation page before the actual quiz begins.  I would like to see that have a header file that would be customizable with the potential for instructions and such.

I personally wouldn't want a header file that is visible throughout the quiz.  Although I can see how some would want such a thing.  

So in I think there should be two headers.  One for a confirmation start page, and another that is optionally shown throughout the quiz.  The confirmation start page should perhaps only be shown if it is a timed quiz.

**Comment by drgrice1** [comment]
It would be good for now to make the grading that is displayed in the quiz and on the homework sets page consistent with the grade shown in the grades page and with what is submitted if LTI grade passback is used.  Both the homework sets page and the gateway quiz module do not allow problems to have a value of zero, and the grade page and LTI grade passback (via WeBWorK::Utils) do allow problems with zero value.

**Comment by Alex-Jordan** [comment]
I am proposing that the existing set header database field for an assignment be utilized. For that, no need to expand the database tables or build a new page. We would just edit the Gateway template and generator to pull whatever set header file is assigned to the problem set and display it on quiz pages.

A confirmation page and information would be good. It's just a larger project, that's all. A new database field that only applies to quizzes for the information to display, and the construction of the page itself.

**Comment by drgrice1** [comment]
I don't think that a confirmation page is any larger of a project than adding in quiz information.

It seems that you are reserving the set_header database column for the purpose you desire, and that is where the notion that a new database column is needed for the other purpose that I propose.

However, a header for the confirmation page (and a database column) is not necessarily needed.  Just a general confirmation page to warn the student that they are about to start a timed exam would be sufficient for me (and probably for most people).
